processor/resourcedetectionprocessor/internal/aws/ecs/ecs.go

```diff
@@ -0,0 +1,166 @@
+	attr.InsertString("cloud.infrastructure_service", "ECS")
+	attr.InsertString("aws.ecs.task.arn", tmdeResp.TaskARN)
+	attr.InsertString("aws.ecs.task.family", tmdeResp.Family)
+
+	// TMDE returns the the short name or ARN, so we need to parse out the short name from ARN if applicable
+	cluster := parseCluster(tmdeResp.Cluster)
+	attr.InsertString("aws.ecs.cluster", cluster)
+
+	region, account := parseRegionAndAccount(tmdeResp.TaskARN)
+	if account != "" {
+		attr.InsertString(conventions.AttributeCloudAccount, account)
+	}
+
+	if region != "" {
+		attr.InsertString(conventions.AttributeCloudRegion, region)
+	}
+
+	// The Availability Zone is not available in all Fargate runtimes
+	if tmdeResp.AvailabilityZone != "" {
+		attr.InsertString(conventions.AttributeCloudZone, tmdeResp.AvailabilityZone)
+	}
+
+	// The launch type and log data attributes are only available in TMDE v4
+	switch lt := strings.ToLower(tmdeResp.LaunchType); lt {
+	case "ec2":
+		attr.InsertString("aws.ecs.launchtype", "EC2")
+
+	case "fargate":
+		attr.InsertString("aws.ecs.launchtype", "Fargate")
+	}
+
+	selfMetaData, err := d.provider.fetchContainerMetaData(tmde)
+
+	if err != nil || selfMetaData == nil {
+		return res, err
+	}
+
+	logAttributes := [4]string{"aws.log.group.names", "aws.log.group.arns", "aws.log.stream.names", "aws.log.stream.arns"}
+
+	for i, attribVal := range getValidLogData(tmdeResp.Containers, selfMetaData, account) {
+		if attribVal.Len() > 0 {
+			ava := pdata.NewAttributeValueArray()
+			ava.SetArrayVal(attribVal)
+			attr.Insert(logAttributes[i], ava)
+		}
+	}
+
+	return res, nil
+}
+
+func getTmdeFromEnv() string {
+	var tmde string
+	if tmde = strings.TrimSpace(os.Getenv(tmde4EnvVar)); tmde == "" {
+		tmde = strings.TrimSpace(os.Getenv(tmde3EnvVar))
+	}
+
+	return tmde
+}
+
+func parseCluster(cluster string) string {
+	parts := strings.Split(cluster, "/")
+	if len(parts) > 1 {
+		return parts[1]
+	}
+
+	return cluster
+}
+
+// Parses the AWS Account ID and AWS Region from a task ARN
+// See: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-account-settings.html#ecs-resource-ids
+func parseRegionAndAccount(taskARN string) (region string, account string) {
+	parts := strings.Split(taskARN, ":")
+	if len(parts) >= 5 {
+		return parts[3], parts[4]
+	}
+
+	return "", ""
+}
+
+// Filter out non-normal containers, our own container since we assume the collector is run as a sidecar,
+// "init" containers which only run at startup then shutdown (as indicated by the "KnownStatus" attribute),
+// containers not using AWS Logs, and those without log group metadata to get the final lists of valid log data
+// See: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4.html#task-metadata-endpoint-v4-response
+func getValidLogData(containers []Container, self *Container, account string) [4]pdata.AnyValueArray {
+	lgn := pdata.NewAnyValueArray()
```

Comment:
While go often contracts variables, I think these could really use some more descriptive names to prevent accidental but :)

Reply:
Will expand